planutils/__init__.py

```diff
@@ -1,31 +1,47 @@
 
 import argparse, os
 
+from planutils import settings
+
 
 def setup():
 
-    assert not_setup_yet(), "Error: planutils is already setup. Remove ~/.planutils to reset (warning: all cached planners will be lost)."
+    assert not_setup_yet(), "Error: planutils is already setup. Remove ~/.planutils to reset (warning: all cached packages will be lost)."
+
+    CUR_DIR = os.path.dirname(os.path.abspath(__file__))
 
     print("\nCreating ~/.planutils...")
     os.mkdir(os.path.join(os.path.expanduser('~'), '.planutils'))
     os.mkdir(os.path.join(os.path.expanduser('~'), '.planutils', 'bin'))
-    os.mkdir(os.path.join(os.path.expanduser('~'), '.planutils', 'bin', 'images'))
+
+    settings.save({
+        'installed': []
+    })
+
+    os.symlink(os.path.join(CUR_DIR, 'packages'),
+               os.path.join(os.path.expanduser('~'), '.planutils', 'packages'))
 
     print("Adding bin folder to path (assuming ~/.bashrc exists)...")
-    os.system('echo \'export PATH="$HOME/.planutils/bin:$PATH"\' >> ~/.bashrc')
+    os.system("echo 'export PLANUTILS_PREFIX=\"~/.planutils\"' >> ~/.bashrc")
+    os.system("echo 'export PATH=\"$PLANUTILS_PREFIX/bin:$PATH\"' >> ~/.bashrc")
 
-    print("Installing planner scripts...")
-    from planutils.planner_installation import PLANNERS
-    for p in PLANNERS:
+    print("Installing package scripts...")
+    from planutils.package_installation import PACKAGES
+    for p in PACKAGES:
         script  = "#!/bin/bash\n"
```

Comment:
I think it might work as-is with a 2-step upgrade path.

1. Upgrading through pip is viable (see [here](https://packaging.python.org/tutorials/installing-packages/#upgrading-packages)). So `pip3 install --upgrade planutils`

2. Setup wouldn't work because of the assumption things haven't been done (error is thrown, and you don't want to be creating directories again). Overwriting just the write stuff in `~/.planutils` could be done with `planutils --upgrade`

Reply:
Is there a way to execute some of your own code triggered by `pip3 install --upgrade`? This seems like a common problem that many pip packages would face, so maybe there is a common solution? If not, the 2-step upgrade would work.

I still have a slight aversion to storing complex bash scripts in python code and overwriting files but I cannot tell you exactly why. It just feels more fragile to me. but I see not technical reason why it wouldn't work.

Reply:
Can't seem to find a firm answer on this. Not the easiest to test...

I've created #15 and will proceed with the 2-step option for now. As for storing the bash, I suppose using a templated file somewhere (using Jinja or the like) would be the proper way. Maybe for v1.0 ;)